src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,52 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleConfirm }) {
+    super({ popupSelector });
+    this._handleConfirm = handleConfirm;
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__button_delete-card"
+    );
+    this._defaultButtonText = this._confirmButton.textContent;
+    this.setEventListeners();
+  }
+
+  ///
+  setButtonText(newText) {
+    this._confirmButton.textContent = newText;
+  }
+
+  resetButtonText() {
+    this._confirmButton.textContent = this._defaultButtonText;
+  }
+  ///
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._confirmButton.addEventListener("click", (event) => {
+      event.preventDefault();
+      this.setButtonText("Saving...");
+      this._handleConfirm();
+      const result = this._handleConfirm;
```

Comment:
`      const result = this._handleConfirm;
`


You code here doesn't work. Everything is in `index.js`